scalyr_agent/builtin_monitors/docker_monitor.py

```diff
@@ -456,6 +483,9 @@ def __update_checkpoints( self ):
         to file.
         """
 
+        if self._use_raw_logs:
```

Comment:
I'm always wary where we add in return points in the middle of a method. I would suggest we instead move the code below into a method like `__watch_container_logs` and then only invoke that here if `!self._use_raw_logs`.